File: BmiApi/BmiApi/Models/BMIRecord.cs
// Models/BMIRecord.cs
using System;

namespace BmiApi.Models
{
    public class BMIRecord
    {
        public long BMIRecordId { get; set; }
        public string? Username { get; set; }
        public double HeightCm { get; set; }
        public double WeightKg { get; set; }
        public double? Bmi { get; set; }
        public long UserId { get; set; }
        public User? User { get; set; }
        public DateTime? DateCreated { get; set; } = DateTime.Now;
    }
} 

File: flutter_bmi_calculator/lib/models/bmi_record.dart
class BmiRecord {
  final int? id;
  final String username;
  final double heightCm;
  final double weightKg;
  final double? bmi;

  BmiRecord({
    this.id,
    required this.username,
    required this.heightCm,
    required this.weightKg,
    required this.bmi,
  });

  factory BmiRecord.fromJson(Map<String, dynamic> json) {
    return BmiRecord(
      id: json['id'],
      username: json['username'],
      heightCm: json['heightCm'],
      weightKg: json['weightKg'],
      bmi: json['bmi'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'username': username,
      'heightCm': heightCm,
      'weightKg': weightKg,
    };
  }
}
